### DISCUSSION TITLE
(Error) (CellFields) RBCCelltype requested but it does not exist

### DISCUSSION THREAD

**Discussion by cerice1995** [UvaCsl/HemoCell#28] [General discussion]
To calibrate parameters, I want to run an RBC generated from an STL through the single cell stretch code. I include the needed STL and XML files which seem to work fine since the lines initialising the particle run successfully (lines 85-94 of the .cpp file, also copied below). However, an error is produced when the RBCCelltype is later requested and "doesn't exist." I think line 109 of the .cpp is requesting the information. Does anyone know why an initialised cell type would not be found? 

------ (lines 85-94)
  // Initialise cells particles
  hemocell.initializeCellfield();
  hemocell.addCellType<RbcHighOrderModel>("RBC_RING_FROM_STL", MESH_FROM_STL);

  // Define any output fields of interest to be written to HDF5 output. 
  hemocell.setOutputs("RBC_RING_FROM_STL", {OUTPUT_POSITION, OUTPUT_TRIANGLES, OUTPUT_FORCE,
                              OUTPUT_FORCE_VOLUME, OUTPUT_FORCE_BENDING,
                              OUTPUT_FORCE_LINK, OUTPUT_FORCE_AREA,
                              OUTPUT_FORCE_VISC});
  hemocell.setFluidOutputs({OUTPUT_VELOCITY, OUTPUT_FORCE});
----- (lines 108-111) 
  // Setting up the stretching
  auto rbc_field = (*hemocell.cellfields)["RBC_RING_FROM_STL"]; // this is likely the error line 
  unsigned n_forced_lsps = 1 + 6;
  HemoCellStretch cellStretch(*rbc_field, n_forced_lsps, stretch_force_si);

[log file.pdf](https://github.com/UvaCsl/HemoCell/files/7709026/log.file.pdf)
[cpp file.pdf](https://github.com/UvaCsl/HemoCell/files/7709027/cpp.file.pdf)

**Comment by MaxvdKolk** [comment]
Does your application only fail with your custom cell `RBC_RING_FROM_STL` or does it also fail with custom cell shapes as used in the examples discussed in the documentation pages: [custom cells](https://www.hemocell.eu/user_guide/advanced_cases/custom_cells.html)?

**Reply by cerice1995** [reply]
I can successfully run examples/cell_shapes. 

I have not tried uploading the cell shapes used in the example into the stretchCell file.
